### DISCUSSION TITLE
Table filter select with options from DB loaded multiple times

### DISCUSSION THREAD

**Discussion by maxacarvalho** [filamentphp/filament#6168] [General]
Hi,

I have a resource that has a filter that gets its options from a table in the DB:
```php
...
use Filament\Tables\Filters\Filter as TableFilter;

...
->filters([
    TableFilter::make(Quote::COMPANY_ID)
        ->label(Str::formatTitle(__('quote.company_id')))
        ->form([
            Select::make(Quote::COMPANY_ID)
                ->label(Str::formatTitle(__('quote.company_id')))
                ->options(Company::all()->pluck(Company::CODE_BRANCH_AND_BRANCH, Company::ID)),
        ])
        ->query(function (Builder $query, array $data): Builder {
            return $query->when(
                $data[Quote::COMPANY_ID],
                fn (Builder $query, int $companyId): Builder => $query->where(Quote::COMPANY_ID, '=', $companyId)
            );
        }),
])
```

Everything works fine with that but the debugbar tells me that `1799` models are being loaded:
<img width="561" alt="image" src="https://user-images.githubusercontent.com/45166618/230946195-e4f14537-e291-447e-a4f7-23b50341bf7c.png">

The `companies` table has only `78` records, so that number, `1794`, doesn't make much sense to me.

Any thoughts? Perhaps I should implement that filter differently?

Thanks!

**Comment by danharrin** [comment]
Try `->options(fn () => Company::all()->pluck(Company::CODE_BRANCH_AND_BRANCH, Company::ID)),`

**Reply by maxacarvalho** [reply]
That worked, thank you!
